moveit_simple/src/moveit_simple.cpp

```diff
@@ -619,46 +671,157 @@ void OnlineRobot::execute(const std::string traj_name, bool collision_check)
+      ROS_ERROR_STREAM("Collision detected in the [" << traj_name << "]");
       throw cd;
     }
   }
   else
   {
-    ROS_ERROR_STREAM("Trajectoy " << traj_name << " not found");
+    ROS_ERROR_STREAM("Trajectoy [" << traj_name << "] not found");
     throw std::invalid_argument("No trajectory found named " + traj_name);
   }
 }
 
 
 
+void OnlineRobot::execute(JointTrajectoryType & goal,
+                          bool collision_check)
+{
+  std::lock_guard<std::recursive_mutex> guard(m_);
+
+  const double TIMEOUT_SCALE = 1.25;  //scales time to wait for action timeout.
+  bool success = false;
+  bool isStatecolliding = false;
+
+  try
+  {
+    int collision_count = 0;
+    for (std::size_t i = 0; i < goal.trajectory.points.size(); i++)
+    {
+      if( (collision_check) && isInCollision(goal.trajectory.points[i].positions ) )
+      {
+        collision_count++;
+        isStatecolliding = true;
+      }
+    }
+
+    if ( !isStatecolliding )
+    {
+
+      // Modify the speed of execution for the trajectory based off of the speed_modifier_
+      for (std::size_t i = 0; i < goal.trajectory.points.size(); i++)
+      {
+        goal.trajectory.points[i].time_from_start *= (1.0/speed_modifier_);
+      }
+
+      ros::Duration traj_time =
+          goal.trajectory.points[goal.trajectory.points.size()-1].time_from_start;
+      ros::Duration timeout(TIMEOUT_SCALE * traj_time.toSec());
+      if (action_.sendGoalAndWait(goal, timeout) == actionlib::SimpleClientGoalState::SUCCEEDED)
+      {
+        ROS_INFO_STREAM("Successfully executed Joint Trajectory ");
+      }
+      else
+      {
+        ROS_ERROR_STREAM("Joint Trajectory failed to execute.. check input Plan");
+        throw ExecutionFailureException("Execution failed for Joint Trajectory");
+      }
+    }
+    else
+    {
+      ROS_ERROR_STREAM("Collision detected at " << collision_count << 
+                       " points for Joint Trajectory");
+      throw CollisionDetected("Collision detected while interpolating ");
+    }
+  }
+  catch(CollisionDetected &cd)
+  {
+    ROS_ERROR_STREAM("Collision detected in trajectory ");
+    throw cd;
+  }
+}
+
+
+
+void Robot::reconfigureRequest(moveit_simple_dynamic_reconfigure_Config &config, uint32_t level)
+{
+  params_.fromConfig(config);
+  if (params_.speed_modifier > 0.0)
+  {
+    setSpeedModifier(params_.speed_modifier);
+  }
+  else
+  {
+    ROS_WARN_STREAM("Speed modifier should be a positive nunber but it is: " <<
+                    params_.speed_modifier);
+  }
+}
+
+
+
+void Robot::setSpeedModifier(const double speed_modifier)
+{
+  if (speed_modifier <= 1.0 && speed_modifier > 0.0)
+  {
+    speed_modifier_ = speed_modifier;
+  }
+  else if ((speed_modifier > 1.0))  
+  {
+    speed_modifier_ = 1.0;
+    ROS_WARN_STREAM("Clamping Speed from " << speed_modifier_ <<
+                    " to max_speed: [1.0]");
+  }
+  else
+  {
+    speed_modifier_ = 0.1;
```

Comment:
I recommend a lower limit....like 0.01

Reply:
I have changed this param, but I would recommend against this.. the simple reason being that a speed of 0.01 is soo soo slow that, more often than not, the program crashes due to some problem in max timeout or thread or something similar.